crates/sui-source-validation/src/tests.rs

```diff
@@ -659,6 +691,14 @@ async fn copy_upgraded_package<'s>(
     toml = toml.replace("$RUNTIME_ID", &runtime_id.to_string());
     tokio::fs::write(dst.join("Move.toml"), toml).await?;
 
+    // Copy Move.lock file if it exists, performing replacements
+    let lock_file = src.join("Move.lock");
+    if lock_file.exists() {
+        let mut toml = tokio::fs::read_to_string(lock_file).await?;
+        toml = toml.replace("$COMPILER_VERSION", CURRENT_COMPILER_VERSION);
+        tokio::fs::write(dst.join("Move.lock"), toml).await?;
+    }
```

Comment:
For tests we pin the Move.lock compiler to the current version (so we are not downloading prior versions). Still getting clarity on [supporting that in CI](https://mysten-labs.slack.com/archives/C03GKUEA5PF/p1705084602180559), but we can move forward with merge and fix forward while functionality is gated by `VERIFIED_SOURCE_BUILD`